### ISSUE TITLE
`/etc/default/docker` and `/etc/init/docker.conf` post-start don't work well together

### ISSUE THREAD

**Issue by krallin** [chef-cookbooks/docker#691]
If you set the following attribute: `default[:docker][:bind_socket] = 'unix:///var/run/docker.sock'`

Then the `/etc/default/docker` file contains:

```
# Docker Upstart and SysVinit configuration file

# Customize location of Docker binary (especially for development testing).
DOCKER="/usr/bin/docker"


# Use DOCKER_OPTS to modify the daemon startup options.
DOCKER_OPTS=" --host=unix:///var/run/docker.sock -r"



# If you need Docker to use an HTTP proxy, it can also be specified here.

# This is also a handy place to tweak where Docker's temporary files go.
```

In turn, this breaks the `post-start` script in `/etc/init/docker.conf`, since that script does **not** wait for the socket to become available if there's a `--host` directive in the file (which is arguably a bit brittle):

```
# Don't emit "started" event until docker.sock is ready.
# See https://github.com/docker/docker/issues/6647
post-start script
        DOCKER_OPTS=
        if [ -f /etc/default/$UPSTART_JOB ]; then
                . /etc/default/$UPSTART_JOB
        fi
        if ! printf "%s" "$DOCKER_OPTS" | grep -qE -e '-H|--host'; then
                while ! [ -e /var/run/docker.sock ]; do
                        initctl status $UPSTART_JOB | grep -q "stop/" && exit 1
                        echo "Waiting for /var/run/docker.sock"
                        sleep 0.1
                done
                echo "/var/run/docker.sock is up"
        fi
end script
```

And therefore actually does emit the "started" event before `docker.sock` is ready :(, which means other jobs that are waiting on Docker to start (with `start on started docker`) will start before Docker is ready, and may crash if they try to access the socket before it's ready.

**Comment by chasebolt** [comment]
You are running a very old version of this cookbook as this is no longer attribute driven. Nonetheless, this is an issue even in the current version and a fix will be going up soon.

**Comment by chasebolt** [comment]
fix released in v2.6.4

**Comment by krallin** [comment]
Ugh, I'm sorry, I should have double-checked the version I was running first. Thanks a lot for the quick fix, though! Glad to know this will work when we update!
